salt/utils/mac_utils.py

```diff
@@ -358,15 +358,21 @@ def _available_services(refresh=False):
 
                 except Exception:
                     # If plistlib is unable to read the file we'll need to use
-                    # the system provided plutil program to do the conversion
+                    # the system provided plutil program to attempt conversion
+                    # from binary.
                     cmd = '/usr/bin/plutil -convert xml1 -o - -- "{0}"'.format(
                         true_path)
-                    plist_xml = __salt__['cmd.run'](cmd)
-                    if six.PY2:
-                        plist = plistlib.readPlistFromString(plist_xml)
-                    else:
-                        plist = plistlib.loads(
-                            salt.utils.stringutils.to_bytes(plist_xml))
+                    try:
+                        plist_xml = __salt__['cmd.run'](cmd)
+                        if six.PY2:
+                            plist = plistlib.readPlistFromString(plist_xml)
+                        else:
+                            plist = plistlib.loads(
+                                salt.utils.stringutils.to_bytes(plist_xml))
+                    except Exception:
```

Comment:
We usually don't like bare excepts. Do you know the specific type(s) of exceptions that can be expected here?

Reply:
Hah, I'm with you on that. I actually put that in there because the previous block used `Exception`. I thought that was a little strange. I'll push some changes to specify the actually expected exceptions tomorrow.